internal/upload/upload_test.go

```
@@ -0,0 +1,724 @@
+	body := []byte("x")
+	s := mustCreate(t, m, []FileDecl{{Path: "ok.flac", Size: 1}}, CreateOptions{})
+	writeAll(t, m, s.ID, s.Files[0].ID, body, 1)
+
+	// Tamper: rewrite the manifest with a traversal path.
+	mp := m.manifestPath(root, s.ID)
+	var doc sessionDoc
+	if err := readJSONFile(mp, &doc); err != nil {
+		t.Fatal(err)
+	}
+	doc.Files[0].RelPath = "../escaped.flac"
+	b, _ := json.Marshal(doc)
```

golangci-lint (2.12.2)

[error] Error return value of `encoding/json.Marshal` is not checked: unsafe type `time.Time` found

(errchkjson)